src/blob/file.rs

```diff
@@ -112,7 +157,17 @@ impl File {
     }
 
     pub(crate) async fn fsyncdata(&self) -> IOResult<()> {
-        let compl = self.ioring.fdatasync(&*self.no_lock_fd);
-        compl.await
+        if let Some(ref ioring) = self.ioring {
+            let compl = ioring.fdatasync(&*self.no_lock_fd);
+            compl.await
+        } else {
+            let (tx, rx) = channel();
```

Comment:
I think could be as well as designated func

Reply:
Done